s3prl/upstream/unispeechsat_hug/expert.py

```diff
@@ -0,0 +1,40 @@
+from typing import List
+
+import torch
+import torch.nn as nn
+from torch import Tensor
+from transformers import AutoFeatureExtractor, UniSpeechSatModel
+
+SAMPLE_RATE = 16000
+
+
+class UpstreamExpert(nn.Module):
+    def __init__(self, ckpt: str = None, model_config: str = None, **kwargs):
+        super().__init__()
+
+        self.processor = AutoFeatureExtractor.from_pretrained(ckpt)
+        self.model = UniSpeechSatModel.from_pretrained(ckpt)
+
+    def get_downsample_rates(self, key: str) -> int:
+        return 320
+
+    def forward(self, wavs: List[Tensor]):
+        device = wavs[0].device
+        processor_outputs = self.processor(
+            [wav.cpu().numpy() for wav in wavs],
+            return_tensors="pt",
+            sampling_rate=SAMPLE_RATE,
+            padding="longest",
+        )
+        attention_mask = processor_outputs.get("attention_mask", None)
+        if isinstance(attention_mask, torch.Tensor):
+            attention_mask = attention_mask.to(device)
```

Comment:
Hi @bagustris!

Specifically I am concerning about this part, which I find it spends time more than `self.model()`. Can you also double check this phenomenon? (I use `time.time`)

Reply:
Now I am inspecting this, will report here...